client/src/App/Components/Mood/MoodIcon.js

```diff
@@ -0,0 +1,50 @@
+import "./MoodIcon.css";
```

Comment:
### 1. Let's refactor ```selectMood``` and ```unSelectMood``` to reduce repeated code

You have two functions, ```selectMood``` and ```unSelectMood```, which both set a div as a state:
The ```selectMood``` div:
```
      <div
        class="selected"
        onClick={() => {
          props.setState(0);
        }}
      >
        {props.image}
      </div>
```

The ```unSelectMood``` div: 

```
      <div
        onClick={() => {
          props.setState(props.number);
        }}
      >
        {props.image}
      </div>
```

Notice there is some repeated code here. The only things that are changing are the ```className``` and the value of the ```state``` state. 

Instead of passing an entire div into the ```selectMood``` and ```unSelectMood``` functions, a more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) approach would be to rename ```mood``` to ```selected```, and set the className as that state inside a div in the component's return statement, like this:

```
  return (
<div>
      <div
        className={selected}
      >
        {props.image}
      </div>
      <p>{props.state}</p>
</div>
  );
```
### 2. Try to replace ```onClick``` with ```useEffect``` inside ```selectMood``` and ```unSelectMood```

If you make the changes outlined above, you'll reduce repetition in your ```selectMood``` and ```unSelectMood``` functions. Good job! :) 

There's a bit more repetition that we can get rid of inside these two functions: namely,  the ```onClick``` event listeners. You can replace these with a ```useEffect``` that will listen for when the ```selected``` state changes and run ```props.setState()```, like this:

```
  useEffect(() => {
    if (props.selected === "") props.setState("");
    else if (props.selected === "selected") props.setState(props.number);
  }, [props.selected]);
```

Reply:
Refactored, at least partially. Pushed.